dart/example/monster_my_game.sample_generated.dart

```diff
@@ -0,0 +1,456 @@
+
+  static const Color Red = const Color._(0);
+  static const Color Green = const Color._(1);
+  static const Color Blue = const Color._(2);
+  static get values => [Red,Green,Blue,];
+
+  static const fb.Reader<Color> reader = const _ColorReader();
+
+  @override
+  String toString() {
+    return 'Color{value: $value}';
+  }
+}
+
+class _ColorReader extends fb.Reader<Color> {
+  const _ColorReader();
+
+  @override
+  int get size => 1;
+
+  @override
+  Color read(fb.BufferContext bc, int offset) =>
+      new Color.fromValue(const fb.Int8Reader().read(bc, offset));
+}
+
+class EquipmentTypeId {
+  final int value;
+  const EquipmentTypeId._(this.value);
+
+  factory EquipmentTypeId.fromValue(int value) {
+    if (value == null) return null;
+    if (value < minValue || maxValue < value) {
+      throw new RangeError.range(value, minValue, maxValue);
+    }
+    return values[value - minValue];
+  }
+
+  static const int minValue = 0;
+  static const int maxValue = 1;
+  static bool containsValue(int value) => minValue <= value && value <= maxValue;
+
+  static const EquipmentTypeId NONE = const EquipmentTypeId._(0);
+  static const EquipmentTypeId Weapon = const EquipmentTypeId._(1);
+  static get values => [NONE,Weapon,];
+
+  static const fb.Reader<EquipmentTypeId> reader = const _EquipmentTypeIdReader();
+
+  @override
+  String toString() {
+    return 'EquipmentTypeId{value: $value}';
+  }
+}
+
+class _EquipmentTypeIdReader extends fb.Reader<EquipmentTypeId> {
+  const _EquipmentTypeIdReader();
+
+  @override
+  int get size => 1;
+
+  @override
+  EquipmentTypeId read(fb.BufferContext bc, int offset) =>
+      new EquipmentTypeId.fromValue(const fb.Uint8Reader().read(bc, offset));
+}
+
+/// Vec3
+abstract class Vec3 {
+  double get x;
+  double get y;
+  double get z;
+
+  factory Vec3(List<int> bytes) {
+    fb.BufferContext rootRef = new fb.BufferContext.fromBytes(bytes);
+    return reader.read(rootRef, 0);
+  }
+
+  static const fb.Reader<Vec3> reader = const _Vec3Reader();
+
+}
+
+class _Vec3Impl implements Vec3 {
+  final fb.BufferContext _bc;
+  final int _bcOffset;
+
+  _Vec3Impl(this._bc, this._bcOffset);
+
+  double _x;
+  @override
+  double get x {
+    _x ??= const fb.Float32Reader().read(_bc, _bcOffset + 0);
+    return _x;
+  }
+
+  double _y;
+  @override
+  double get y {
+    _y ??= const fb.Float32Reader().read(_bc, _bcOffset + 4);
+    return _y;
+  }
+
+  double _z;
+  @override
+  double get z {
+    _z ??= const fb.Float32Reader().read(_bc, _bcOffset + 8);
+    return _z;
+  }
+
+  @override
+  String toString() {
+    return 'Vec3{x: $x, y: $y, z: $z}';
+  }
+}
+
+class _Vec3Reader extends fb.StructReader<_Vec3Impl> {
+  const _Vec3Reader();
+
+  @override
+  int get size => 12;
+
+  @override
+  _Vec3Impl createObject(fb.BufferContext bc, int offset) => 
+    new _Vec3Impl(bc, offset);
+}
+
+class Vec3Builder extends fb.BuilderHelper {
+  final double _x;
```

Comment:
Builder classes in other languages don't actually store anything, they directly write to a buffer. The whole design of FlatBuffer is about avoid object construction / copying, which this ignores.

Reply:
This will store additional references.  See my comments above on `example.dart`

Reply:
Again, as per the above, I don't think any values should be stored. There really is no reason to.

Reply:
I think this makes sense to rework as a simple method then on the `Vec3` class.  This will be a substantial change though.

Reply:
Coding is done, working on upating the docs/tests.